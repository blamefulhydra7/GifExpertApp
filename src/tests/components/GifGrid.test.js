import React from 'react';
import {shallow} from 'enzyme'
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en <GifGrid/>', () => {
    const category = 'Goku';

    test('Debe de mostrar el componente correctamente', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });
        const wrapper = shallow(<GifGrid category={category}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe mostrar items cuando cargan las imÃ¡genes', () => {
        const gifs = [{
            id: 'ABC',
            url: 'http://localhost/cualquiercosa.jpg',
            title: 'Cualquier cosa',
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });

        const wrapper = shallow(<GifGrid category={category}/>);

        // expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });
});