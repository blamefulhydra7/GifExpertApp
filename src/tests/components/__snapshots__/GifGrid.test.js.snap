// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid/> Debe de mostrar el componente correctamente 1`] = `
<Fragment>
  <h3
    className="animate__fadeIn animate__animated"
  >
    Goku
  </h3>
  <p
    className="animate__flash animate__animated"
  >
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
